### PR TITLE
Convert to Gradle and ditch Travis for GitHub actions.

### PR DESCRIPTION
**Pull request by vanniktech**
References #10

Took things over from crossword

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          body: ${{ steps.release_notes.outputs.release_notes }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish Artifacts
+        run: ./gradlew publish
+        if: github.ref == 'refs/heads/trunk'
+        env:
+          ORG_GRADLE_PROJECT_mavenCentralUsername: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
+          ORG_GRADLE_PROJECT_mavenCentralPassword: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
```
Comment (vanniktech) [inline .github/workflows/build.yaml:42]: You need to set those 2 (if they aren't already set)

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        if: startsWith(github.ref, 'refs/tags/')
+        uses: ffurrer2/extract-release-notes@v1
+
+      - name: Create Release
+        if: startsWith(github.ref, 'refs/tags/')
+        uses: softprops/action-gh-release@v1
+        with:
+          body: ${{ steps.release_notes.outputs.release_notes }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (vanniktech) [inline .github/workflows/build.yaml:35]: This one too

Comment (JakeWharton) [reply .github/workflows/build.yaml:35]: This one is set by default https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret

File: .github/workflows/build.yaml (hunk 1 above)
Comment (JakeWharton) [reply .github/workflows/build.yaml:42]: Will do. It's annoying there aren't user-level secrets like we get with orgs.

**Pull request merged**